src/transformers/searchservice/ResultsFactory.ts

```diff
@@ -115,10 +115,18 @@ export class ResultsFactory {
   }
 
   private static fromCustomSource(result: any, index: number): Result {
+    return this.fromCustomSourceHelper(result, index, Source.Custom);
+  }
+
+  private static fromFunctionVertical(result: any, index: number): Result {
+    return this.fromCustomSourceHelper(result, index, Source.FunctionVertical);
+  }
+
+  private static fromCustomSourceHelper(result: any, index: number, source: Source): Result {
```

Comment:
Maybe name this like "fromCustomDataHelper"? CustomSource works well for "fromCustomSource", but sounds a bit odd for "fromFunctionVertical"